### PR TITLE
implemented testcase for new object sets using pytest

### PR DESCRIPTION
**Pull request by ar-india**
changes -

made entire testcase Thread safe. Pytest provides a mechanism by which tests can be run in parallell using multiple thread workers. hence i decided to implement syncronization of common functions such as create_volume(),delete_volume() etc.
Although as of today we will not leverage threading as it won't provide much of a performance difference. but in future if more tests are added and ,say,the tests take half an hour to complete , then we can use threading to acheive faster results.
for running pytest with multiple thread worker we need to install a new module called "pytest-xdist"
testcase related to volume, snapshot, volcoll have few common functionality and they should be made thread safe .
-impleneted testcase for new object sets -

README.md test_access_control_records.py test_arrays.py test_events.py
config.ini test_active_directory_membership.py test_audit_log.py test_master_key.py
conftest.py test_alarms.py test_chap_users.py test_application_categories.py test_controllers.py
test_application_servers.py test_disks.py

-Have written the readme file

### CODE REVIEW

**Comment by ar-india** [comment]
-added new testcase for below obect set.

 new file:   tests/setup.cfg
        new file:   tests/test_fibre_channel_configs.py
        new file:   tests/test_fibre_channel_ports_and_sessions.py
        new file:   tests/test_folders.py
        new file:   tests/test_groups.py
        new file:   tests/test_initiator_and_initiatorgroups.py
        new file:   tests/test_jobs.py
        new file:   tests/test_volume_collection.py

**Comment by ar-india** [comment]
-added testcase to cover the below object set

Z:\projects\ansible_nimble\github_checkin\nimble-python-sdk>git commit -s -m "part4 - adding more testcases for nimble python sdk"
[pytest c49d987] part4 - adding more testcases for nimble python sdk
 26 files changed, 1447 insertions(+), 34 deletions(-)
 create mode 100644 tests/conftest.py
 create mode 100644 tests/test_network_configs.py
 create mode 100644 tests/test_network_interfaces.py
 create mode 100644 tests/test_performance_policies.py
 create mode 100644 tests/test_pools.py
 create mode 100644 tests/test_protection_schedules.py
 create mode 100644 tests/test_protection_tempalate.py
 create mode 100644 tests/test_protocol_endpoints.py
 create mode 100644 tests/test_snapshot_collection.py
 create mode 100644 tests/test_snapshots.py

File: tests/.vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,46 @@
+{
```
Comment (ar-india) [inline tests/.vscode/launch.json:1]: launch.json file is being checked in for developers who may need to add new test case or debug existing code using visual studio. this file has the settings necessary to debug python file. one just needs to make sure this folder ".vscode" exists at the same location where the tests folder is

**Comment by ar-india** [comment]
-added testcase for below. with these we have completed all the testcase related to Nimble sdk .
 create mode 100644 tests/.vscode/launch.json
 create mode 100644 tests/__init__.py
 create mode 100644 tests/test_replication_partners.py
 create mode 100644 tests/test_shelves.py
 create mode 100644 tests/test_software_versions.py
 create mode 100644 tests/test_space_domain.py
 create mode 100644 tests/test_subnet.py
 create mode 100644 tests/test_user.py

**Review by suneeth51 — APPROVED** [review]
lgtm...

**Comment by ar-india** [comment]
not able to resolve conflicts .. will raise a new pr with same change

**Pull request closed without merging**